lib/client/mfa/fido.go

```diff
@@ -0,0 +1,208 @@
+package mfa
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/segmentio/aws-okta/lib/client/types"
+
+	log "github.com/sirupsen/logrus"
+
+	u2fhost "github.com/marshallbrekka/go-u2fhost"
+)
+
+const (
+	MaxOpenRetries = 10
+	RetryDelayMS   = 200 * time.Millisecond
+)
+
+var (
+	errNoDeviceFound = fmt.Errorf("no U2F devices found. device might not be plugged in")
+)
+
+// FIDODevice is implementation of MFADevice for SMS
+type FIDODevice struct {
+	userInput     Input
+	codeRequested bool
+	id            string
+}
+
+func (d *FIDODevice) SetId(id string) {
+	d.id = id
+}
+
+func (d *FIDODevice) GetId() string {
+	return d.id
+}
+
+// Supported will check if the mfa config can be used by this device
+func (d *FIDODevice) Supported(factor types.OktaUserAuthnFactor) error {
+	if factor.FactorType == "u2f" && factor.Provider == "FIDO" {
+		d.id = factor.Id
```

Comment:
Is this the only `Supported` that mutates its receiver? That seems odd

Reply:
It looks like we don't need this as we call `SetId()` from okta.go:314.

Reply:
I've removed SetId/GetId and mutation in Supported for the FIDO implementation.